### PR TITLE
notation 1.0.0 (new formula)

### PR DESCRIPTION
**Pull request by mintbomb27**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/n/notation.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  depends_on "go" => :build
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    project = "github.com/notaryproject/notation"
+    ldflags = %W[
+      -s -w
+      -X #{project}/internal/version.Version=v#{version}
+      -X #{project}/internal/version.GitCommit=#{Utils.git_short_head}
+      -X #{project}/internal/version.BuildMetadata=
```
Comment (shizhMSFT) [inline Formula/n/notation.rb:19]: It is better to indicate that it is built by homebrew.
```suggestion
      -X #{project}/internal/version.BuildMetadata=Homebrew
```

File: Formula/n/notation.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    ]
+    system "go", "build", *std_go_args(output: bin/"notation", ldflags: ldflags), "./cmd/notation"
+  end
```
Comment (shizhMSFT) [inline Formula/n/notation.rb:22]: `generate_completions_from_executable`?

File: Formula/n/notation.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class Notation < Formula
+  desc "CLI tool to sign and verify OCI artifacts and container images"
+  homepage "https://notaryproject.dev/"
+  url "https://github.com/notaryproject/notation.git",
+    tag:      "v1.0.0",
+    revision: "80e3fc4e2eeb43ac00bc888cf41101f5c56f1535"
```
Comment (shizhMSFT) [inline Formula/n/notation.rb:6]: I'm wondering what's the difference between this one and
```
  url "https://github.com/notaryproject/notation/archive/refs/tags/v1.0.0.tar.gz"
  sha256 "c7e3a154cb132d17c72c9954e2c2144b5dd1781af69e9bfe25ec29916e2f01be"
```

Comment (mintbomb27) [reply Formula/n/notation.rb:6]: So I used the git method to add the GitCommit in the Build Flags, to use `#{Utils.git_short_head}`

File: Formula/n/notation.rb (hunk 1 above)
Comment (mintbomb27) [reply Formula/n/notation.rb:19]: Sure! Adding.

File: Formula/n/notation.rb (hunk 2 above)
Comment (mintbomb27) [reply Formula/n/notation.rb:22]: Adding for default three shells.

File: Formula/n/notation.rb (hunk 3 above)
Comment (chenrui333) [reply Formula/n/notation.rb:6]: yeah, that is the catch, I would also see you can use `#{tap.user}` to fill that commit, the commit is more relevant to upstream build, not to homebrew build though.

**Review by chenrui333 — APPROVED** [review]

File: Formula/n/notation.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  head "https://github.com/notaryproject/notation.git", branch: "main"
+
+  depends_on "go" => :build
+
+  def install
+    project = "github.com/notaryproject/notation"
+    ldflags = %W[
+      -s -w
+      -X #{project}/internal/version.Version=v#{version}
+      -X #{project}/internal/version.GitCommit=#{tap.user}
```
Comment (shizhMSFT) [inline Formula/n/notation.rb:16]: I'm not able to find the documentation for `#{tap.user}`. What's the value of it? May I have the link to the documentation?

**Review by p-linnane — APPROVED** [review]

File: Formula/n/notation.rb (hunk 4 above)
Comment (mintbomb27) [reply Formula/n/notation.rb:26]: Don't know why but it's value is `Homebrew`, and value of `tap` is `homebrew/core`. Looks like it is the repository's or tap's name.

Comment (shizhMSFT) [reply Formula/n/notation.rb:26]: We can just remove line 16. If `GitCommit` is empty, `notation version` will show
```
notation version
Notation - a tool to sign and verify artifacts.

Version:     v1.0.0+Homebrew
Go version:  go1.21.0
```
without `Git commit:`.

See code: https://github.com/notaryproject/notation/blob/v1.0.0/cmd/notation/version.go#L42-L44

Comment (mintbomb27) [reply Formula/n/notation.rb:26]: But, `Utils.git_short_head` provides the git commit. So why not use it?

**Pull request merged**